modules/apps/foundation/portal-configuration/portal-configuration-test-util/src/main/java/com/liferay/portal/configuration/test/util/ConfigurationTemporarySwapper.java

```diff
@@ -65,6 +58,13 @@ public ConfigurationTemporarySwapper(
 
 		_oldProperties = _configuration.getProperties();
 
+		_serviceServiceTracker = ServiceTrackerFactory.open(
+			bundle, serviceClass);
+
+		Object service = _serviceServiceTracker.waitForService(5000);
+
+		_validateService(serviceClass, service, pid);
+
 		_serviceListener = new ConfigurationServiceListener(
```

Comment:
@drewbrokke it makes sense now, I will resend with restoring the order, but the null check is still useless. In case of throwing the exception, the ConfigurationTemporarySwapper was never created, therefore the close is never called, there is no chance that the close method can see a null _configuration object.